src/plone/restapi/tests/test_blocks_serializer.py

```diff
@@ -189,3 +192,95 @@ def test_serialize_subblocks_transformers(self):
         href = value["1"]["data"]["blocks"]["2"]["blocks"]["3"]["href"]
 
         self.assertEqual(href[0], self.portal.doc1.absolute_url())
+
+    def test_internal_link_serializer(self):
+        """test_internal_link_serializer."""
+        doc_uid = IUUID(self.portal["doc1"])
+        resolve_uid_link = {
+            "@id": "../resolveuid/{}".format(doc_uid),
+            "title": "Welcome to Plone",
+        }
+        blocks = {
+            "2caef9e6-93ff-4edf-896f-8c16654a9923": {
+                "@type": "slate",
+                "plaintext": "this is a slate link inside some text",
+                "value": [
+                    {
+                        "children": [
+                            {"text": "this is a "},
+                            {
+                                "children": [
+                                    {"text": ""},
+                                    {
+                                        "children": [{"text": "slate link"}],
+                                        "data": {
+                                            "link": {
+                                                "internal": {
+                                                    "internal_link": [resolve_uid_link]
+                                                }
+                                            }
+                                        },
+                                        "type": "a",
+                                    },
+                                    {"text": ""},
+                                ],
+                                "type": "strong",
+                            },
+                            {"text": " inside some text"},
+                        ],
+                        "type": "p",
+                    }
+                ],
+            },
+            "6b2be2e6-9857-4bcc-a21a-29c0449e1c68": {"@type": "title"},
+        }
+
+        res = self.serialize(
+            context=self.portal["doc1"],
+            blocks=blocks,
+        )
+
+        value = res["2caef9e6-93ff-4edf-896f-8c16654a9923"]["value"]
+        link = value[0]["children"][1]["children"][1]
+        resolve_link = link["data"]["link"]["internal"]["internal_link"][0]["@id"]
+
+        self.assertTrue(resolve_link == self.portal.absolute_url() + "/doc1")
+
+    def test_simple_link_serializer(self):
+        doc_uid = IUUID(self.portal["doc1"])
+        resolve_uid_link = "../resolveuid/{}".format(doc_uid)
```

Comment:
@tiberiuichim can you elaborate why we have two different kinds of links in Slate? Pls, excuse my ignorance on Slate internals here.

Reply:
"simple link" is the "new" link plugin that I've prepared, the one that uses the default Volto [AddLinkForm](https://github.com/plone/volto/blob/master/src/components/manage/AnchorPlugin/components/LinkButton/AddLinkForm.jsx)

The "internal link" is the current implementation that existing slate deployments use, kept for backwards compatibility.

Reply:
@tisto please not that it needs this merged into Volto: https://github.com/plone/volto/pull/2421